src/js/cart.js

```diff
@@ -49,5 +50,38 @@ function calculateTotal(cartItems) {
   return total;
 }
 
-renderCartContents();
+function renderCartContents() {
+  const cartItems = getLocalStorage("so-cart") || [];
+  const htmlItems = cartItems.map((item) => cartItemTemplate(item));
+  const productList = document.querySelector(".product-list");
+
+  productList.innerHTML = htmlItems.join("");
+
+  if (cartItems.length > 0) {
+    let cartFooter = document.querySelector(".cart-footer");
+    cartFooter.classList.remove("hide");
+
+    let totalAmount = calculateTotal(cartItems);
 
+    let totalElement = document.getElementById("totalAmount");
+    totalElement.innerText = "Total: $" + totalAmount.toFixed(2);
+
+    const removeButtons = document.querySelectorAll(".remove-item");
+    removeButtons.forEach((button) => {
+      button.addEventListener("click", removeFromCart);
+    });
+  }
+}
+
+function removeFromCart(event) {
+  const productIdToRemove = event.target.dataset.id;
+  const updatedCart = getLocalStorage("so-cart").filter(
```

Comment:
No need to have a default here because this function should _never_ get called if the cart is null. Nice.